index/test/rtree/rtree_with_strategies.cpp

```diff
@@ -0,0 +1,156 @@
+// Boost.Geometry Index
+// Unit Test
+
+// Copyright (c) 2025 Adam Wulkiewicz, Lodz, Poland.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#include <vector>
+
+#define BOOST_GEOMETRY_INDEX_DETAIL_EXPERIMENTAL_PREDICATES
```

Comment:
Is it still experimental? When can we promote it to "general"?

Reply:
Here it's only for `touches` and only to test if it compiles. I can't remember exactly but I think it's experimental because it makes little sense to calculate whether or not bounding boxes touch. It could be removed I guess.

There is also experimental path predicate, aka distance along linestring/segment, aka ray query which could be useful but it probably requires more thinking.

Reply:
OK - I understand - but you'll probably have the best understanding of this code.

If it makes no sense - we should remove it?

Reply:
Now that I think about it. The R-tree also works for segments so maybe `touches(segment, segment)` could be useful? But this should be done in a different PR anyway.